study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/StudyBuilder.java

```diff
@@ -376,6 +379,20 @@ private void insertStudyDetails(Handle handle, long studyId) {
         }
     }
 
+    private void insertStudySupportedLanguages(Handle handle, long studyId) {
+        String studyGuid = handle.attach(JdbiUmbrellaStudy.class).findById(studyId).getGuid();
+        cfg.getStringList("studySupportedLanguages").forEach(
+                isoCode -> {
+                    Long langCodeId = handle.attach(JdbiLanguageCode.class).getLanguageCodeId(isoCode);
+                    Optional.ofNullable(langCodeId).orElseThrow(
+                            () -> new DDPException("Could not find language using code: " + langCodeId)
```

Comment:
It's nice to check that it's a valid language code, but we don't necessarily have to do it ourselves. We're doing a subquery for the `language_code_id` in `addSupportedLanguage()`. So if it's not a valid lang code, the SQL will fail. We can just let that sql exception propagate up.

Reply:
Yes, you're right, this definitely seems redundant. However, I faced that SQL exception once and it seemed a bit obscure to me - I had to spend some mental energy to figure out what went wrong. I'd leave it as-is as a courtesy to other folks that can stumble upon it in the future.  And, given the nature of StudyBuilder, the cost of it is negligible. Are you ok with that?

Reply:
Sure, that's reasonable.